model/src/test/test_random_key.py

```diff
@@ -70,44 +90,91 @@ def test_rng_keys_produce_correct_samples():
     # set up singular epidemiological process
 
     # set up base models for testing
-    model_01 = create_test_model()
-    model_02 = create_test_model()
-    model_03 = create_test_model()
-    model_04 = create_test_model()
-    model_05 = create_test_model()
-
+    models = [create_test_model() for _ in range(5)]
+    n_timepoints_to_simulate = [30] * len(models)
     # sample only a single model and use that model's samples
     # as the observed_infections for the rest of the models
     with npro.handlers.seed(rng_seed=np.random.randint(1, 600)):
-        model_01_samp = model_01.sample(n_timepoints_to_simulate=30)
+        model_sample = models[0].sample(
+            n_timepoints_to_simulate=n_timepoints_to_simulate[0]
+        )
+    obs_infections = [model_sample.observed_infections] * len(models)
+    rng_keys = [jr.key(54), jr.key(54), None, None, jr.key(74)]
 
     # run test models with the different keys
-    models = [model_01, model_02, model_03, model_04, model_05]
-    rng_keys = [jr.key(54), jr.key(54), None, None, jr.key(74)]
-    obs_infections = [model_01_samp.observed_infections] * len(models)
+
     for elt in list(zip(models, obs_infections, rng_keys)):
-        sample_test_model(*elt)
+        run_test_model(*elt)
 
+    prior_predictive_list = [
+        prior_predictive_test_model(*elt)
+        for elt in list(zip(models, n_timepoints_to_simulate, rng_keys))
+    ]
+
+    posterior_predictive_list = [
+        posterior_predictive_test_model(*elt)
+        for elt in list(zip(models, n_timepoints_to_simulate, rng_keys))
+    ]
     # using same rng_key should get same run samples
     assert np.array_equal(
-        model_01.mcmc.get_samples()["Rt"][0],
-        model_02.mcmc.get_samples()["Rt"][0],
+        models[0].mcmc.get_samples()["Rt"][0],
+        models[1].mcmc.get_samples()["Rt"][0],
+    )
+
+    assert np.array_equal(
```

Comment:
Suggest [`numpy.testing.assert_array_equal`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) throughout

Reply:
Eventually we might move to [`chex.assert_trees_all_equal`](https://chex.readthedocs.io/en/latest/api.html#chex.assert_trees_all_equal) if we decide to depend on `chex` for testing (see #223)

Reply:
addressed in https://github.com/CDCgov/multisignal-epi-inference/pull/218/commits/8e154ce3270afbea7f1d177483b1cd78ab110510